pkgs/by-name/fi/firefly-iii/package.nix

```diff
@@ -0,0 +1,69 @@
+{ stdenvNoCC
+, lib
+, fetchzip
+, nixosTests
+, php83
+, dataDir ? "/var/lib/firefly-iii"
+, settings ? {}
+}:
+
+let
+  version = "6.1.13";
+  phpPackage = php83;
+  fireflyEnvVars = lib.generators.toKeyValue {
+    mkKeyValue = lib.flip lib.generators.mkKeyValueDefault "=" {
+      mkValueString = v: with builtins;
+        if isInt             v then toString v
+        else if isString     v then "\"${v}\""
+        else if true  ==     v then "True"
+        else if false ==     v then "False"
+        else throw "unsupported type ${typeOf v}: ${(lib.generators.toPretty {}) v}";
+    };
+  };
+  firefly-iii-env = (fireflyEnvVars settings);
+in
+
+stdenvNoCC.mkDerivation {
+  pname = "firefly-iii";
+  inherit version;
+
+  src = fetchzip {
+    url = "https://github.com/firefly-iii/firefly-iii/releases/download/v${version}/FireflyIII-v${version}.zip";
+    hash = "sha256-97ml3b19IJd70xGDnvBwLUJSw3Ms4BL3qGQ62KLTyFE=";
+    stripRoot = false;
+  };
+
+  buildInputs = [ phpPackage ];
+
+  dontBuild = true;
+  dontFixup = true;
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out
+    cp -r * $out/
+    runHook postInstall
+  '';
+
+  passthru = {
+    inherit phpPackage;
+    tests = nixosTests.firefly-iii;
+  };
+
+  postInstall = ''
+    rm -R $out/storage $out/bootstrap/cache
+    ln -s ${dataDir}/storage $out/storage
+    ln -s ${dataDir}/cache $out/bootstrap/cache
+    cat <<EOF > $out/.env
+      ${firefly-iii-env}
+    EOF
```

Comment:
Whie the env file is easily moved into the module, I'm less sure about how to go about moving the storage and cache symlink directories to a wrapper package. `symlinkJoin` would lead to a million different symlinks given the sheer number of packages in the `vendor` directory. Should I just copy the unwrapped package wholesale?

Reply:
@SuperSandro2000 I tried doing this using a variety of methods, but the service doesn't work properly if the source files are symlinked in a wrapper. The only way I've found that does work is the method used by bookstack, which is to add a symlink to the data directory in the original unwrapped package.

Reply:
If we can't easily do it, then we keep it like this.